File: Bridge/Resources/Math.js
    // @source Math.js

    var math = {
        divRem: function(a, b, result) {
            var remainder = a % b;
            result.v = remainder;
            return (a - remainder) / b;
        },

        round: function(n, d, rounding) {
            var m = Math.pow(10, d || 0);
            n *= m;
            var sign = (n > 0) | -(n < 0);
            if (n % 1 === 0.5 * sign) {
                var f = Math.floor(n);
                return (f + (rounding === 4 ? (sign > 0) : (f % 2 * sign))) / m;
            }

            return Math.round(n) / m;
        }
    };

    Bridge.Math = math;

File: Compiler/Lua/CoreSystem.Lua/CoreSystem/Math.lua
local System = System

local Math = math

Math.bigMul = function(a, b) 
    return a * b 
end

Math.divRem = function(a, b) 
    local remainder = a % b;
    return (a - remainder) / b, remainder
end

Math.round = function(n, d, rounding) 
    local m = Math.pow(10, d or 0);
    n = n * m
    local sign = n > 0 and 1 or -1
    if n % 1 == 0.5 * sign then 
        local f = Math.floor(n)
        return (f + (rounding == 1 and (sign > 0) or (f % 2 * sign))) / m
    end
    return Math.floor(n + 0.5) / m
end

Math.sign = function(v) 
    return v == 0 and 0 or (v > 0 and 1 or -1) 
end

Math.trunc = System.trunc
System.Math = Math






